internal/tweets/edit.go

```diff
@@ -3,18 +3,17 @@ package tweets
 import (
 	"database/sql"
 	"encoding/json"
-	"fmt"
 	"net/http"
 
 	"Twitter_like_application/internal/services"
 
-	"github.com/go-playground/validator/v10"
 	"github.com/gorilla/mux"
 )
 
-type EditTweetRequest struct {
-	Text string `json:"text" validate:"required,checkTweetText"`
+type editTweetRequest struct {
+	Text string `json:"text" validate:"required,text"`
 	Visibility
+	createTweetRequest
```

Comment:
this is wrong, better use the same type if there are no differences or use an alias

Reply:
now you have a struct with Text field and embedded Visibility fields and embedded Text and Visibility fields again. This is too much of unneeded complexity and very confusing. Can lead to mistakes easily.